src/api/app/controllers/trigger_workflow_controller.rb

```diff
@@ -15,6 +15,7 @@ class TriggerWorkflowController < ApplicationController
 
   before_action :set_token
   before_action :validate_token_type
+  before_action :validate_scm_vendor
```

Comment:
```suggestion
  before_action :validate_scm_vendor
  before_action :set_token
  before_action :validate_token_type
```
Wouldn't it be better to check for the request headers in first instance, and then set the token?